components/backup/src/endpoint.rs

```diff
@@ -438,6 +505,13 @@ impl<E: Engine, R: RegionInfoProvider> Runnable<Task> for Endpoint<E, R> {
     }
 }
 
+impl<E: Engine, R: RegionInfoProvider> RunnableWithTimer<Task, ()> for Endpoint<E, R> {
+    fn on_timeout(&mut self, timer: &mut Timer<()>, _: ()) {
+        self.pool.borrow_mut().check_active();
+        timer.add_task(Duration::from_millis(IDLE_THREADPOOL_CHECK_INTERVAL), ());
```

Comment:
I'd like to extend the interval to avoid create and destroy threads too frequently.

Reply:
this interval is for periodically check whether thread pool is idle, we can change another constant IDLE_THREADPOOL_DURATION to do this. Btw, how much time you think is proper.

Reply:
I think 30min should be fine.

Reply:
OK